app/channels/student_tests_channel.rb

```diff
@@ -0,0 +1,28 @@
+class StudentTestsChannel < ApplicationCable::Channel
+  def subscribed
+    course = Course.find_by(id: params[:course_id])
+    role = Role.find_by(user: current_user, course: course)
+    if role.nil?
+      reject
+      return
+    end
+
+    assignment = Assignment.find_by(id: params[:assignment_id])
```

Comment:
For these queries it would be good to use associations to further restrict the models that can be accessed. E.g., `course.assignments.find_by(...)`